Frameworks/CoreGraphics/CGDataConsumer.mm

```diff
@@ -16,56 +16,119 @@
 
 #import <StubReturn.h>
 #import <CoreGraphics/CGDataConsumer.h>
+#import <CFCppBase.h>
+#import <Starboard/SmartTypes.h>
+#import "CGDataConsumerInternal.h"
+
+#pragma region __CGDataConsumer
+
+static const CGDataConsumerCallbacks sc_CGDataConsumerCFDataCallbacks{ // CGDataConsumerPutBytesCallback
+                                                                       [](void* info, const void* data, size_t size) {
+                                                                           CFDataAppendBytes(static_cast<CFMutableDataRef>(info),
+                                                                                             (const UInt8*)data,
+                                                                                             size);
+                                                                           return size;
+                                                                       },
+
+                                                                       // CGDataConsumerReleaseInfoCallback
+                                                                       [](void* info) { CFRelease(static_cast<CFTypeRef>(info)); }
+};
+
+static const CGDataConsumerCallbacks sc_CGDataConsumerCFURLCallbacks{ // CGDataConsumerPutBytesCallback
+                                                                      [](void* info, const void* data, size_t size) -> size_t {
+                                                                          auto writeStream = woc::MakeAutoCF<CFWriteStreamRef>(
```

Comment:
?!?!?!
creating a new `CFWriteStream` is identical to writing the data to the new file again every time...

You have to make the write stream the shared context.